addons/web/static/src/core/utils/numbers.js

```diff
@@ -177,6 +177,9 @@ export function humanNumber(number, options = { decimals: 0, minDigits: 1 }) {
  * @returns {string}
  */
 export function formatFloat(value, options = {}) {
+    if (value === false || value === null) {
```

Comment:
Look at the docstring. This function is there to format numbers, `false` and `null` aren't numbers. So it shouldn't been called with those arguments. The `formatFloat` method in formatters is a bit more tolerant as a float field's value can be false. However, I don't know where the `null` comes from.

Reply:
It's called from the Float field...
```js
# /web/static/src/views/fields/float/float_field.js
return formatFloat(this.value, { digits: this.digits, humanReadable: false });
```

> The formatFloat method in formatters is a bit more tolerant as a float field's value can be false

So I guess the float formatter used in the Float field is the wrong one ?

Reply:
Exactly. The mistake has been introduced here <https://github.com/odoo/odoo/pull/133824>. I think that this isn't the only place where the wrong formatter has been used. I'll take a deeper look tomorrow.